pixel/image.go

```diff
@@ -43,6 +43,40 @@ func NewImage[T Color](width, height int) Image[T] {
 	}
 }
 
+// NewImageFromBytes creates a new image of the given size using an existing data slice of bytes.
+func NewImageFromBytes[T Color](width, height int, buf []byte) Image[T] {
```

Comment:
Should this return `(Image[T], error)` instead of panicking?

Reply:
That is the current interface, which I was not trying to change in this PR.